website/content/docs/nia/usage/zscaler-application-segments.mdx

```diff
@@ -0,0 +1,404 @@
+- create Terraform files for the defined tasks, and 
+- connect to Consul.
+
+```shell-session
+$ consul-terraform-sync start -config-file=tasks.hcl
+[INFO] v0.6.0 (7f0d7ba)
+[INFO] (driver.terraform) installing terraform to path '/root/nia'
+[INFO] (driver.terraform) successfully installed terraform
+[INFO] (templates.hcltmpl) evaluating dynamic configuration for "thunder"
+## ...
+[INFO] running Terraform command: /root/nia/terraform init -no-color -force-copy -input=false -lock-timeout=0s -backend=true -get=true -upgrade=false -lock=true -get-plugins=true -verify-plugins=true
+Initializing modules...
+Downloading a10networks/service-group-sync-nia/thunder 0.1.6 for slb_auto_config...
+- slb_auto_config in .terraform/modules/slb_auto_config
+
+Initializing the backend...
+
+Successfully configured the backend "consul"! Terraform will automatically
+use this backend unless the backend configuration changes.
+## ...
+Terraform has been successfully initialized!
+[INFO] running Terraform command: /root/nia/terraform workspace new -no-color zpa_application_segment
+Workspace "zpa_application_segment" already exists
+[INFO] running Terraform command: /root/nia/terraform workspace select -no-color slb_auto_config
+Switched to workspace "slb_auto_config".
+[INFO] running Terraform command: /root/nia/terraform apply -no-color -auto-approve -input=false -var-file=terraform.tfvars -var-file=providers.tfvars -lock=true -parallelism=10 -refresh=true
+## ...
+
+Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
+[INFO] (ctrl) task completed zpa_application_segment
+[INFO] (ctrl) all tasks completed once
+[INFO] (cli) running controller in daemon mode
+[INFO] (api) starting server at '8558'
+```
+
+CTS continuously runs as a daemon and automatically executes the `zpa_application_segment` task when it detects changes to the `web` Consul service.
+
+## Verify NIA automation workflow
+
+Verify the workflow by registering new instances of the `web` service in Consul.
+
+Create a service configuration file with the following details named `web2.hcl`. The only difference between this configuration and `web.hcl` is the service ID is set to `web2`.
+
+<CodeBlockConfig filename="web2.hcl">
+
+```hcl
+service {
+  name = "web"
+  id = "web2"
+  tags = ["web"]
+  port = 80
+  check {
+    id = "web"
+    name = "web TCP check"
+    tcp = "localhost:80"
+    interval = "15s"
+    timeout = "1s"
+  }
+}
+```
+
+</CodeBlockConfig>
+
+Then, register the service with Consul.
+
+```shell-session
+$ consul services register web2.hcl
+```
+
+Once CTS detects the new Consul service, it will automatically trigger the `zpa_application_segment` task, as defined in the configuration file. The task uses Terraform to update the ZPA application segment domain entries to include the new Consul service instances. Instead, if you de-registered the Consul service, CTS will trigger the ZPA application segment domain entries to remove the servers. 
```

Comment:
```suggestion
Once CTS detects the new Consul service, it will automatically trigger the `zpa_application_segment` task, as defined in the configuration file. The task uses Terraform to update the ZPA application segment domain entries to include the new Consul service instances. Instead, if you de-registered the Consul service, CTS will update the ZPA application segment domain entries to remove the servers. 
```

Reply:
We used trigger above for the module itself. `update` here seems more correct.